openapi/spec.go

```diff
@@ -305,3 +308,24 @@ type Tag struct {
 	Name        string `json:"name" yaml:"name"`
 	Description string `json:"description,omitempty" yaml:"description,omitempty"`
 }
+
+type SecuritySchemes struct {
+	Type             string                 `json:"type,omitempty" yaml:"type,omitempty"`
+	Scheme           string                 `json:"scheme,omitempty" yaml:"scheme,omitempty"`
+	Description      string                 `json:"description,omitempty" yaml:"description,omitempty"`
+	In               string                 `json:"in,omitempty" yaml:"in,omitempty"`
+	Name             string                 `json:"name,omitempty" yaml:"name,omitempty"`
+	OpenIdConnectUrl string                 `json:"openIdConnectUrl,omitempty" yaml:"openIdConnectUrl,omitempty"`
+	Flows            map[string]*OAuthFlows `json:"flows,omitempty" yaml:"flows,omitempty"`
+}
+
+type OAuthFlows struct {
```

Comment:
This object is invalid. See the difference between `OAuth Flows Object` and `OAuth Flow Object` in the spec.

Reply:
Where can i find `OAuth Flow Object` I was using this, https://swagger.io/docs/specification/authentication/oauth2/

thanks

Reply:
@hmajid2301 
https://swagger.io/specification/#oauth-flows-object
https://swagger.io/specification/#oauth-flow-object

Reply:
Is there another way to do this? Rather than using a map. A way to specify the four different flow types ? 

```go
flows: map[string]*OAuthFlows{
  "implicit": &OAuthFlows{
    AuthorizationURL: "https://example.com/api/oauth/dialog",
    Scopes: map[string]string {
        "write:pets": "modify pets in your account",
        "read:pets": "read your pets"
      }
  }
}
```

Reply:
Sorry for the delay.
Why not using a struct type?
```
type OAuthFlows struct {
  Implicit *OAuthFlow `json:"implicit,omitempty"`
  ...
}
```

Reply:
No worries, I'm sure you are busy.

Thanks that's a very good point :). 
I forgot about `omitempty`, still to used to Typescript and unions etc.

One final thing I will have to add a function to allow users to add security/scopes to each endpoint as well correct ? For this Issue ? 

Thanks

Reply:
> One final thing I will have to add a function to allow users to add security/scopes to each endpoint as well

I don't think a `fizz.OperationOption` is necessary in that particular usecase. The Security Requirement Object is defined on the root OpenAPI object, it would make more sense to let the user configure these part of the spec manually, in the same way the Server Object can be configured, like so:
```go
	f := fizz.NewFromEngine(r)
	if url != "" {
		f.Generator().SetServers([]*openapi.Server{
			&openapi.Server{
				URL:             url,
				Description: description,
			},
		})
	}
```
So instead, a function similar to `SetServers` could be added to change the value of the `security` field.

Reply:
How can we add `security` (scopes) to each specific endpoint ? 

```yaml
paths:
  /pet:
    post:
      tags:
      - "pet"
      summary: "Add a new pet to the store"
      description: ""
      operationId: "addPet"
      consumes:
      - "application/json"
      - "application/xml"
      produces:
      - "application/xml"
      - "application/json"
      parameters:
      - in: "body"
        name: "body"
        description: "Pet object that needs to be added to the store"
        required: true
        schema:
          $ref: "#/definitions/Pet"
      responses:
        "405":
          description: "Invalid input"
      security:
      - petstore_auth:
        - "write:pets"
        - "read:pets"
```
In the example above `petstore_auth` ? Thanks

Reply:
Ah, you're speaking of the `security` field of the Operation Object: https://swagger.io/specification/#operation-object

For that, the `Operation` type must be update to include the field `security` which is a Security Requirement Object: https://swagger.io/specification/#operation-object, which could be translated to a `map[string][]string`.

A `fizz.OperationOption` would be necessary to override the security requirements per operation, similar to the existing option funcs.

Reply:
yeh sorry i should've been more clear with my wording. Ok I'll add to the PR as well ?